examples/resnet/cifar_input.py

```diff
@@ -67,42 +70,35 @@ def build_input(data, batch_size, dataset, train):
     Raises:
       ValueError: When the specified dataset is not supported.
     """
-    images_constant = tf.constant(data[0])
-    labels_constant = tf.constant(data[1])
     image_size = 32
     depth = 3
     num_classes = 10 if dataset == "cifar10" else 100
-    image, label = tf.train.slice_input_producer([images_constant,
-                                                  labels_constant],
-                                                 capacity=16 * batch_size)
-    if train:
+    images, labels = data
+    num_samples = images.shape[0] - images.shape[0] % batch_size
+    dataset = tf.contrib.data.Dataset.from_tensor_slices(
+        (images[:num_samples], labels[:num_samples]))
+
+    def map_train(image, label):
         image = tf.image.resize_image_with_crop_or_pad(image, image_size + 4,
                                                        image_size + 4)
         image = tf.random_crop(image, [image_size, image_size, 3])
         image = tf.image.random_flip_left_right(image)
         image = tf.image.per_image_standardization(image)
-        example_queue = tf.RandomShuffleQueue(
-            capacity=16 * batch_size,
-            min_after_dequeue=8 * batch_size,
-            dtypes=[tf.float32, tf.int32],
-            shapes=[[image_size, image_size, depth], [1]])
-        num_threads = 16
-    else:
+        return (image, label)
+
+    def map_test(image, label):
         image = tf.image.resize_image_with_crop_or_pad(image, image_size,
                                                        image_size)
         image = tf.image.per_image_standardization(image)
-        example_queue = tf.FIFOQueue(
-            3 * batch_size,
-            dtypes=[tf.float32, tf.int32],
-            shapes=[[image_size, image_size, depth], [1]])
-        num_threads = 1
-
-    example_enqueue_op = example_queue.enqueue([image, label])
-    tf.train.add_queue_runner(tf.train.queue_runner.QueueRunner(
-        example_queue, [example_enqueue_op] * num_threads))
-
-    # Read "batch" labels + images from the example queue.
-    images, labels = example_queue.dequeue_many(batch_size)
+        return (image, label)
+
+    dataset = dataset.map(map_train if train else map_test)
+    dataset = dataset.batch(batch_size)
+    dataset = dataset.repeat()
+    if train:
+        dataset = dataset.shuffle()
```

Comment:
Don't you need a `buffer_size` for `shuffle()`? https://www.tensorflow.org/versions/r1.3/api_docs/python/tf/contrib/data/Dataset#shuffle

Reply:
Nice good catch.

Reply:
I think you should also shuffle before batching.

Reply:
^+1 good catch @rshin